src/SerializeAST.cc

```diff
@@ -0,0 +1,766 @@
+    }
+    else if (!propertyName.variable.empty()) {
+        propertyNameObject.set(SerializeKey::Variable, WrapValueDefinition(propertyName.variable));
+    }
+
+    return propertyNameObject;
+}
+
+// Forward declarations
+sos::Object WrapTypeSection(const mson::TypeSection& typeSection);
+
+
+const sos::String TypeSectionToString(const mson::TypeSection& section)
+{
+    switch (section.klass) {
+        case mson::TypeSection::BlockDescriptionClass:
+            return sos::String("blockDescription");
+
+        case mson::TypeSection::MemberTypeClass:
+            return sos::String("memberType");
+
+        case mson::TypeSection::SampleClass:
+            return sos::String("sample");
+
+        case mson::TypeSection::DefaultClass:
+            return sos::String("default");
+
+        default:
+            return sos::String();
+    }
+    return sos::String();
+}
+
+sos::String AssetRoleToString(const AssetRole& role)
+{
+    std::string str;
+
+    switch (role) {
+        case BodyExampleAssetRole:
+            str = "bodyExample";
+            break;
+
+        case BodySchemaAssetRole:
+            str = "bodySchema";
+            break;
+
+        default:
+            break;
+    }
+
+    return sos::String(str);
+}
+
+sos::String ElementClassToString(const Element::Class& element)
+{
+    std::string str;
+
+    switch (element) {
+        case Element::CategoryElement:
+            str = "category";
+            break;
+
+        case Element::CopyElement:
+            str = "copy";
+            break;
+
+        case Element::ResourceElement:
+            str = "resource";
+            break;
+
+        case Element::DataStructureElement:
+            str = "dataStructure";
+            break;
+
+        case Element::AssetElement:
+            str = "asset";
+            break;
+
+        default:
+            break;
+    }
+
+    return sos::String(str);
+}
+
+sos::Object WrapNamedType(const mson::NamedType& namedType)
+{
+    sos::Object namedTypeObject;
+
+    // Name
+    namedTypeObject.set(SerializeKey::Name, WrapTypeName(namedType.name));
+
+    // Type Definition
+    namedTypeObject.set(SerializeKey::TypeDefinition, WrapTypeDefinition(namedType.typeDefinition));
+
+    // Type Sections
+    namedTypeObject.set(SerializeKey::Sections, 
+                        WrapCollection<mson::TypeSection>()(namedType.sections, WrapTypeSection));
+
+    return namedTypeObject;
+}
+
+
+sos::Object WrapKeyValue(const KeyValuePair& keyValue)
+{
+    sos::Object keyValueObject;
+
+    // Name
+    keyValueObject.set(SerializeKey::Name, sos::String(keyValue.first));
+
+    // Value
+    keyValueObject.set(SerializeKey::Value, sos::String(keyValue.second));
+
+    return keyValueObject;
+}
+
+sos::Object WrapMetadata(const Metadata& metadata)
+{
+    return WrapKeyValue(metadata);
+}
+
+sos::Object WrapHeader(const Header& header)
+{
+    return WrapKeyValue(header);
+}
+
+sos::Object WrapReference(const Reference& reference)
```

Comment:
Why are the above 4 functions here? They should be after all the MSON wrapping functions are done.